VideoSearchWithAutoMLVision/AutoML_Vision_Classifier_PowerSkill/powerskill/vi.py

```diff
@@ -0,0 +1,138 @@
+import logging
+
+import requests
+
+
+class VideoIndexerAPI():
+    def __init__(self, vi_api_key, vi_location, vi_account_id):
+        self.vi_api_key = vi_api_key
+        self.vi_location = vi_location
+        self.vi_account_id = vi_account_id
+        self.access_token = None
+
+    def get_access_token(self):
+        """
+        Here we get an access token for the video indexer instance
+        :return:
+        """
+        logging.info('Getting video indexer access token...')
+        headers = {
+            'Ocp-Apim-Subscription-Key': self.vi_api_key
+        }
+
+        params = {
+            'allowEdit': 'true'
+        }
+        access_token_req = requests.get(
+            'https://api.videoindexer.ai/auth/{loc}/Accounts/{acc_id}/AccessToken'.format(
+                loc=self.vi_location,
+                acc_id=self.vi_account_id
+            ),
+            params=params,
+            headers=headers
+        )
+
+        access_token = access_token_req.text[1:-1]
+        logging.info('Access Token successfully retrieved')
+        self.access_token = access_token
+        return access_token
+
+    def get_thumbnail(self, video_id, thumbnail_id):
+        """
+        Get a thumbnail from the video
+        :param video_id: Id of the video
+        :param thumbnail_id: Id of the thumbnail
+        :return: The image
+        """
+        logging.info('Getting video thumbnail..')
+
+        headers = {
+            'Ocp-Apim-Subscription-Key': self.vi_api_key
+        }
+
+        params = {
+            'location': self.vi_location,
+            'accountId': self.vi_account_id,
+            'videoId': video_id,
+            'thumbnailId': thumbnail_id,
+            'format': 'Jpeg',
+            'accessToken': self.access_token
```

Comment:
Prefer transferring access token in headers instead of query params. (true everywhere)

Reply:
Implemented